core/state/database.go

```diff
@@ -181,85 +253,57 @@ type ForkingDB struct {
 	StorageProcessed bool
 }
 
-// ContractCode implements Database
-func (fdb *ForkingDB) ContractCode(addrHash common.Hash, codeHash common.Hash) ([]byte, error) {
-	if fdb.started {
-		return fdb.VerkleDB.ContractCode(addrHash, codeHash)
-	}
-
-	return fdb.cachingDB.ContractCode(addrHash, codeHash)
-}
-
-// ContractCodeSize implements Database
-func (fdb *ForkingDB) ContractCodeSize(addrHash common.Hash, codeHash common.Hash) (int, error) {
-	if fdb.started {
-		return fdb.VerkleDB.ContractCodeSize(addrHash, codeHash)
+// OpenTrie opens the main account trie at a specific root hash.
+func (db *cachingDB) openMPTTrie(root common.Hash) (Trie, error) {
+	tr, err := trie.NewStateTrie(trie.StateTrieID(root), db.db)
+	if err != nil {
+		return nil, err
 	}
-
-	return fdb.cachingDB.ContractCodeSize(addrHash, codeHash)
+	return tr, nil
 }
 
-// CopyTrie implements Database
-func (fdb *ForkingDB) CopyTrie(t Trie) Trie {
-	return fdb.cachingDB.CopyTrie(t)
-}
+func (db *cachingDB) openVKTrie(root common.Hash) (Trie, error) {
 
-func verkleOrTransitionTree(self Trie) Trie {
-	switch t := self.(type) {
-	case *trie.TransitionTrie:
-		return t.Overlay()
-	case *trie.VerkleTrie:
-		return t
-	default:
-		panic("unexpected trie type")
+	payload, err := db.DiskDB().Get([]byte("flat-"))
+	if err != nil {
+		return trie.NewVerkleTrie(verkle.New(), db.db, db.addrToPoint), nil
 	}
-}
 
-// OpenStorageTrie implements Database
-func (fdb *ForkingDB) OpenStorageTrie(stateRoot, addrHash, root common.Hash, self Trie) (Trie, error) {
-	mpt, err := fdb.cachingDB.OpenStorageTrie(stateRoot, addrHash, root, nil)
-	if fdb.started && err == nil {
-		// Return a "storage trie" that is an adapter between the storge MPT
-		// and the unique verkle tree.
-		vkt, err := fdb.VerkleDB.OpenStorageTrie(stateRoot, addrHash, fdb.getTranslation(root), verkleOrTransitionTree(self))
-		if err != nil {
-			return nil, err
-		}
-		return trie.NewTransitionTree(mpt.(*trie.SecureTrie), vkt.(*trie.VerkleTrie), true), nil
+	r, err := verkle.ParseNode(payload, 0)
+	if err != nil {
+		panic(err)
```

Comment:
Avoid panic since the method has the error return parameter
```suggestion
		return nil, fmt.Errorf("parsing node: %s", err)
```